universal-application-tool-0.0.1/app/views/HtmlBundle.java

```diff
@@ -174,6 +185,24 @@ private ContainerTag renderMain() {
     return mainTag;
   }
 
+  private ContainerTag renderModals() {
+    ContainerTag modalContainer =
+        div()
+            .withId("modal-container")
+            .withClasses(BaseStyles.MODAL_CONTAINER)
+            .with(div().withId("modal-glass-pane").withClasses(BaseStyles.MODAL_GLASS_PANE));
+    modals.forEach(
+        modal ->
+            modalContainer.with(
+                div()
+                    .with(
+                        button("x")
+                            .withId(modal.modalId() + "-close")
+                            .withClasses(BaseStyles.MODAL_CLOSE_BUTTON))
+                    .with(modal.getContainerTag())));
+    return modalContainer;
```

Comment:
Yeah, if you used the alternative method you'd just have this instead...

```suggestion
    ContainerTag modalContent = div.withId("modal-content");
    ContainerTag hiddenModals = div().hidden();
    modals.forEach(modal -> hiddenModals.with(modal.getContainerTag()));
    modalContainer.with(
                div()
                    .with(
                        button("x")
                            .withId(modal.modalId() + "-close")
                            .withClasses(BaseStyles.MODAL_CLOSE_BUTTON))
                    .with(modalContent);
    return modalContainer;
```